src/game_battlealgorithm.cpp

```diff
@@ -764,7 +758,7 @@ void Game_BattleAlgorithm::Skill::Apply() {
 	}
 	else {
 		if (first_attack) {
-			source->SetSp(source->GetSp() - source->CalculateSkillCost(skill.ID));
+			source->ChangeSp(- source->CalculateSkillCost(skill.ID));
```

Comment:
You can do it like here in the first_attack-if instead (but in Normal::Apply ^^)